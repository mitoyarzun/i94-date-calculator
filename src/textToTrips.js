import { TravelTypes } from './Trips';

function isDate (date) {
  return date.match(/[0-9]{4}-[0-9]{2}-[0-9]{2}/);
}

function isTravelType (type) {
  return type === TravelTypes.Arrival || type === TravelTypes.Departure;
}

function isPortOfEntry (port) {
  return port.match(/[A-Z]{3}|Unavailable/);
}

export default function textToTrips (text) {
  const trips = [];
  
  let currentDate = null;
  let currentTravelType = null;
  
  text.split(/\n/).forEach((line) => {
    if (isDate(line)) {
      currentDate = line;
    } else if (isTravelType(line)) {
      currentTravelType = line;
    } else if (isPortOfEntry(line)) {
      trips.push({
        date: currentDate,
        parsedDate: new Date(currentDate),
        travelType: currentTravelType,
        portOfEntry: line,
      });
    }
  });
  
  return trips;
}
