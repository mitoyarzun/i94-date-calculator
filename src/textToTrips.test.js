import textToTrips from './textToTrips';

describe('textToTrips', () => {
    it('should convert text to trips', () => {
        const text = `
1
2018-03-13
Arrival
JFK
2
2018-01-20
Departure
SEA
3
2018-01-03
Arrival
PHO
4
2017-08-17
Departure
CLT
5
2017-02-21
Arrival
SFR
6
2017-02-20
Departure
Unavailable
`;

        expect(textToTrips(text)).toMatchSnapshot();
    });
});
