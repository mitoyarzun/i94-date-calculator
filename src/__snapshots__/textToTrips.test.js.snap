// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`textToTrips should convert text to trips 1`] = `
Array [
  Object {
    "date": "2018-03-13",
    "parsedDate": 2018-03-13T00:00:00.000Z,
    "portOfEntry": "JFK",
    "travelType": "Arrival",
  },
  Object {
    "date": "2018-01-20",
    "parsedDate": 2018-01-20T00:00:00.000Z,
    "portOfEntry": "SEA",
    "travelType": "Departure",
  },
  Object {
    "date": "2018-01-03",
    "parsedDate": 2018-01-03T00:00:00.000Z,
    "portOfEntry": "PHO",
    "travelType": "Arrival",
  },
  Object {
    "date": "2017-08-17",
    "parsedDate": 2017-08-17T00:00:00.000Z,
    "portOfEntry": "CLT",
    "travelType": "Departure",
  },
  Object {
    "date": "2017-02-21",
    "parsedDate": 2017-02-21T00:00:00.000Z,
    "portOfEntry": "SFR",
    "travelType": "Arrival",
  },
  Object {
    "date": "2017-02-20",
    "parsedDate": 2017-02-20T00:00:00.000Z,
    "portOfEntry": "Unavailable",
    "travelType": "Departure",
  },
]
`;
